### PR TITLE
Update day08.md

### PR DESCRIPTION
**Pull request by prinsniels**
Sorry for f* up the initial PR

### CODE REVIEW

**Comment by bishabosha** [comment]
Hey, thanks again for writing this up, but please could you observe the format for the entry points as described in the [schedule](https://github.com/scalacenter/scala-advent-of-code/discussions/348), also we should be getting a solution from @iusildra shortly

**Comment by prinsniels** [comment]
A, sorry. Wasn't aware off that

> Hey, thanks again for writing this up, but please could you observe the format for the entry points as described in the [schedule](https://github.com/scalacenter/scala-advent-of-code/discussions/348), also we should be getting a solution from @iusildra shortly

**Comment by bishabosha** [comment]
> A, sorry. Wasn't aware off that

its fine! I should probably change the text in the website template

File: docs/2023/puzzles/day08.md [hunk 1]
```diff
@@ -6,6 +6,160 @@ import Solver from "../../../../../website/src/components/Solver.js"
...
+  def gcd(a: Long, b: Long): Long =
+    if b == 0 then a else gcd(b, a % b)
+
+  starts
+    .map(state => countStepsUntil(state, instructions, trans, 0, _.endsWith("Z")).toLong)
+    .reduce(lcm) pipe println
+
+object Solution extends App:
+  part1("day08.txt".live)
+  part2("day08.txt".live)
```
Comment (bishabosha) [inline docs/2023/puzzles/day08.md:160]:
```suggestion
```

Comment (bishabosha) [reply docs/2023/puzzles/day08.md:160]: removing as not necessary

File: docs/2023/puzzles/day08.md [hunk 2]
```diff
@@ -6,6 +6,156 @@ import Solver from "../../../../../website/src/components/Solver.js"
...
+  starts
+    .map(state => countStepsUntil(state, instructions, trans, 0, _.endsWith("Z")).toLong)  // for each state find the cycle time
+    .reduce(lcm) pipe println
+```
+
+## final code
+```scala
+package aoc
+package solutions
+
```
Comment (bishabosha) [inline docs/2023/puzzles/day08.md:94]:
```suggestion
```
also can be removed

File: docs/2023/puzzles/day08.md [hunk 3]
```diff
@@ -6,6 +6,156 @@ import Solver from "../../../../../website/src/components/Solver.js"
...
+import scala.util.chaining.*
```
Comment (bishabosha) [inline docs/2023/puzzles/day08.md:95]:
```suggestion
```
don't need to print

File: docs/2023/puzzles/day08.md [hunk 4]
```diff
@@ -6,6 +6,152 @@ import Solver from "../../../../../website/src/components/Solver.js"
...
+    d match
+      case Instr.GoLeft  => network(n)(0)
+      case Instr.GoRight => network(n)(1)
+
+@tailrec
+def countStepsUntil(state: State, instrs: LazyList[Instr], trans: Transition, count: Int, pred: State => Boolean): Int =
+  if pred(state) then count
+  else countStepsUntil(trans(state, instrs.head), instrs.tail, trans, count + 1, pred)
+
+def part1(input: String) =
```
Comment (bishabosha) [inline docs/2023/puzzles/day08.md:128]:
```suggestion
def part1(input: String): Int = 
```
return types

File: docs/2023/puzzles/day08.md [hunk 5]
```diff
@@ -6,6 +6,152 @@ import Solver from "../../../../../website/src/components/Solver.js"
...
+  (n, d) =>
+    d match
+      case aoc.solutions.Instr.GoLeft  => network(n)(0)
+      case aoc.solutions.Instr.GoRight => network(n)(1)
+```
+
+For the predicate tell the function to stop when `STATE == "ZZZ"`
+```scala
+def one(inp: String) =
+  val inpL         = inp.split("\n\n")
```
Comment (bishabosha) [inline docs/2023/puzzles/day08.md:63]:
```suggestion
  val inpL         = input.split("\n\n")
```
match final code

File: docs/2023/puzzles/day08.md [hunk 6]
```diff
@@ -6,6 +6,154 @@ import Solver from "../../../../../website/src/components/Solver.js"
...
+
+  def gcd(a: Long, b: Long): Long =
+    if b == 0 then a else gcd(b, a % b)
+
+  // get all the starting states
+  val starts: Set[State] = network.keySet.filter(_.endsWith("A"))
+
+  starts
+    .map(state => countStepsUntil(state, instructions, trans, 0, _.endsWith("Z")).toLong)  // for each state find the cycle time
+    .reduce(lcm) pipe println
```
Comment (bishabosha) [inline docs/2023/puzzles/day08.md:89]:
```suggestion
    .reduce(lcm)
```
no printing

**Comment by bishabosha** [comment]
@prinsniels I have just applied a few small refactorings to enhance readability with the narrow width of the code blocks

**Pull request merged**